_includes/page-specific-js.lqd

```diff
@@ -0,0 +1,28 @@
+{% comment %}
+    Include a JS file for the given URL
+{% endcomment %}
+{% capture cache %}
+    {% assign parts = include.url | remove: ".html" |remove_first: '/' | split: "/" %}
+    {% assign slug = '' %}
+    {% assign tags = '' %}
+
+    {% for part in parts %}
+        {% assign part = part | slugify %}
+
+        {% if forloop.first == true %}
+            {% assign slug = slug | append: part %}
+        {% else %}
+            {% assign slug = slug | append: '-' | append: part %}
+        {% endif %}
+
+        {% assign filePath = slug | prepend: '/assets/js/' | append: '.js' %}
+
+        {% for static_file in site.static_files %}
+            {% if static_file.path == filePath %}
+                {% capture output %}
+  <script src="{{ filePath | prepend:site.baseurl }}"></script>{% endcapture  %}
```

Comment:
Indenting is as intended. In order to avoid extra indenting where this macro is used, no whitespace is added here.